### ISSUE TITLE
We should also probably make `AlignedType` contain `MaybeUninit` or something like that, as the C code might write padding bytes to it.

### ISSUE THREAD

**Issue by apoelstra** [rust-bitcoin/rust-secp256k1#707]
We should also probably make `AlignedType` contain `MaybeUninit` or something like that, as the C code might write padding bytes to it.

I think it wasn't stable (enough) back when I implemented this logic

_Originally posted by @elichai in https://github.com/rust-bitcoin/rust-secp256k1/pull/635#discussion_r1667327931_

**Comment by Kixunil** [comment]
Indeed, `MaybeUninit` was stabilized quite late relative to both rust-bitcoin birth and my involvement in Rust.

**Comment by tcharding** [comment]
This weeks candidate for worst issue title ever :)

**Comment by apoelstra** [comment]
Yeah, sorry, normally I shorten the result of "reference in new issue" but I got lazy.
